internal/provider/metric_resource.go

```diff
@@ -0,0 +1,357 @@
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	statuspal "terraform-provider-statuspal/internal/client"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var (
+	_ resource.Resource                = &MetricResource{}
+	_ resource.ResourceWithImportState = &MetricResource{}
+)
+
+func NewMetricResource() resource.Resource {
+	return &MetricResource{}
+}
+
+// MetricResource defines the resource implementation.
+type MetricResource struct {
+	client *statuspal.Client
+}
+
+// MetricType represents the type of metric (either "up" or "rt").
+type MetricType string
+
+const (
+	UptimeMetric       MetricType = "up"
+	ResponseTimeMetric MetricType = "rt"
+)
+
+// FeaturedNumber represents the type of number to display for the metric.
+type FeaturedNumber string
+
+const (
+	AvgFeatured  FeaturedNumber = "avg"
+	MaxFeatured  FeaturedNumber = "max"
+	LastFeatured FeaturedNumber = "last"
+)
+
+// MetricResourceModel represents the data model for a metric resource.
+//
+// https://www.statuspal.io/api-docs#tag/Metrics/operation/addMetric
+type MetricResourceModel struct {
+	Subdomain       types.String `tfsdk:"subdomain"`
+	ID              types.Int64  `tfsdk:"id"`
+	Title           types.String `tfsdk:"title"`
+	Unit            types.String `tfsdk:"unit"`
+	Type            types.String `tfsdk:"type"`
+	Enabled         types.Bool   `tfsdk:"enabled"`
+	Visible         types.Bool   `tfsdk:"visible"`
+	RemoteID        types.String `tfsdk:"remote_id"`
+	RemoteName      types.String `tfsdk:"remote_name"`
+	Status          types.String `tfsdk:"status"`
+	LatestEntryTime types.Int64  `tfsdk:"latest_entry_time"`
+	Threshold       types.Int64  `tfsdk:"threshold"`
+	FeaturedNumber  types.String `tfsdk:"featured_number"`
+	Order           types.Int64  `tfsdk:"order"`
+	IntegrationID   types.String `tfsdk:"integration_id"`
+}
+
+func (r *MetricResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_metric"
+}
+
+func (r *MetricResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"id": schema.Int64Attribute{
+				Description: "The unique identifier for the metric.",
+				Computed:    true,
+			},
+			"status": schema.StringAttribute{
+				Description: "The status of the metric.",
+				Computed:    true,
+			},
+			"latest_entry_time": schema.Int64Attribute{
+				Description: "The timestamp for the latest entry of the metric.",
+				Computed:    true,
+			},
+			"order": schema.Int64Attribute{
+				Description: "The order of the metric in the system.",
+				Computed:    true,
+			},
+			"subdomain": schema.StringAttribute{
+				Description: "The subdomain of the metric.",
+				Required:    true,
+			},
```

Comment:
Look at for example the service resource.
We have the `status_page_subdomain` and `id` at the top level and we have a `service` `SingleNestedAttribute`.
Please follow that format.
https://github.com/statuspal/terraform-provider-statuspal/blob/main/internal/provider/service_resource.go#L117-L126